tests/cuda/test_cuda_jagged_vector_view.cpp

```diff
@@ -33,30 +32,28 @@ class cuda_jagged_vector_view_test : public testing::Test {
             vecmem::vector<int>(&m_mem),
             vecmem::vector<int>({12, 13, 14, 15, 16}, &m_mem)
         }, &m_mem),
-        m_data(m_vec, &m_mem)
+        m_data(vecmem::get_data(m_vec, &m_mem))
     {
     }
 };
 
 TEST_F(cuda_jagged_vector_view_test, mutate_in_kernel) {
     doubleJagged(m_data);
 
-    vecmem::jagged_device_vector<int> m_jag(m_data);
-
-    EXPECT_EQ(m_jag.at(0, 0), 2);
-    EXPECT_EQ(m_jag.at(0, 1), 4);
-    EXPECT_EQ(m_jag.at(0, 2), 6);
-    EXPECT_EQ(m_jag.at(0, 3), 8);
-    EXPECT_EQ(m_jag.at(1, 0), 10);
-    EXPECT_EQ(m_jag.at(1, 1), 12);
-    EXPECT_EQ(m_jag.at(2, 0), 14);
-    EXPECT_EQ(m_jag.at(2, 1), 16);
-    EXPECT_EQ(m_jag.at(2, 2), 18);
-    EXPECT_EQ(m_jag.at(2, 3), 20);
-    EXPECT_EQ(m_jag.at(3, 0), 22);
-    EXPECT_EQ(m_jag.at(5, 0), 24);
-    EXPECT_EQ(m_jag.at(5, 1), 26);
-    EXPECT_EQ(m_jag.at(5, 2), 28);
-    EXPECT_EQ(m_jag.at(5, 3), 30);
-    EXPECT_EQ(m_jag.at(5, 4), 32);
+    EXPECT_EQ(m_vec.at(0).at(0), 2);
+    EXPECT_EQ(m_vec.at(0).at(1), 4);
+    EXPECT_EQ(m_vec.at(0).at(2), 6);
+    EXPECT_EQ(m_vec.at(0).at(3), 8);
+    EXPECT_EQ(m_vec.at(1).at(0), 10);
+    EXPECT_EQ(m_vec.at(1).at(1), 12);
+    EXPECT_EQ(m_vec.at(2).at(0), 14);
+    EXPECT_EQ(m_vec.at(2).at(1), 16);
+    EXPECT_EQ(m_vec.at(2).at(2), 18);
+    EXPECT_EQ(m_vec.at(2).at(3), 20);
+    EXPECT_EQ(m_vec.at(3).at(0), 22);
+    EXPECT_EQ(m_vec.at(5).at(0), 24);
+    EXPECT_EQ(m_vec.at(5).at(1), 26);
+    EXPECT_EQ(m_vec.at(5).at(2), 28);
+    EXPECT_EQ(m_vec.at(5).at(3), 30);
+    EXPECT_EQ(m_vec.at(5).at(4), 32);
```

Comment:
After some pondering I decided that it makes more sense to check the values in the "original" jagged vector instead of looking at the contents of `m_data` using another instances of `vecmem::jagged_device_vector`. But let me know if you think otherwise @stephenswat.

Reply:
Works for me, as long as we design everything such that the underlying `std::vector` is the ultimate owner of all the data, and that they are responsible for freeing all of it. That's a fairly natural design anyway, in this case, because `std::vector` doesn't really allow you to release its memory from it. Not without some fairly unpleasant code, anyway.

Reply:
The code already tests `vecmem::jagged_device_vector`'s behaviour in `test_core_jagged_vector_view.cpp`. In this CUDA test I believe we want to test that the CUDA kernel is able to modify the values inside of the "original" `vecmem::jagged_vector` in-situ. That's why I thought the tests would make more sense expressed through the `m_vec` object.

Any other comments about these proposed changes in `vecmem::data::jagged_vector_data`? :thinking: